src/AppShed/Extensions/SpreadsheetBundle/Service/GeoService.php

```diff
@@ -11,33 +11,46 @@
  */
 class GeoService {
 
+    private $geoClient;
+
+    public function __construct(Client $geoClient)
+    {
+        $this->geoClient = $geoClient;
+    }
+
     public function getGeo($address)
     {
-        $guzzle = new Client();
 
-        $address = urlencode($address);
+        try {
+            $geoResponse = $this->geoClient->get(
+                'geocode/json',
+                [
+                    'query' => [
+                        'address' => $address
+                    ]
+                ]
+            );
 
-        $url = "http://maps.google.com/maps/api/geocode/json?sensor=false&address={$address}";
+            if($geoResponse->getStatusCode()==200){
 
-        $res = $guzzle->get($url);
+                $resp = $geoResponse->json();
 
-        if($res->getStatusCode()==200){
+                if ($resp['status']=='OK' && count($resp['results'])) {
 
-            $resp = $res->json();
+                    $lat = $resp['results'][0]['geometry']['location']['lat'];
+                    $lon = $resp['results'][0]['geometry']['location']['lng'];
 
-            if ($resp['status']=='OK') {
+                    return ['lon'=>$lon, 'lat'=>$lat];
```

Comment:
notice that you could just `return $resp['results'][0]['geometry']['location'];` its basically the same, and less code!